js/pantry.js

```diff
@@ -115,73 +70,109 @@ function renderFromStorage(storageData) {
   if (currentLocalStorage) {
-    // check for duplicate item names
-    let quantityBefore;
-    const checkDuplicate = currentLocalStorage.some((elem) => {
-      quantityBefore = elem.quantity;
-      return elem.name.toLowerCase() === values[0].toLowerCase();
-    });
-    if (checkDuplicate) {
-      // remove from local storage to update the quantity then add back
-
-      // since the input was a duplicate, delete from local storage
-      const updatedQuantity = deleteItemFromStorage(values);
-      //
-      // get the sum of the previous quantity and the new quantity entered
-      values[1] = parseInt(values[1]) + parseInt(quantityBefore);
-
-      // add the new item to local storage with the updated quantity
-      updatedQuantity.push(new PantryItem(...values)); // push a new PantryItem with the quanitities added together
-      setLocalStorage("pantry", updatedQuantity);
-      renderFromStorage(getLocalStorage("pantry"));
-
-      return;
+
+    let fruitArr = [];
+
+    for (let item of currentLocalStorage) {
+      fruitArr.push(item.name.toLowerCase());
+
     }
+    
+    //checks for duplicates from user input and prompts to add quantity instead
+    
+    let fruitInput = event.target.food.value.toLowerCase()
+    if (fruitArr.includes(fruitInput)) {
+
+      for (let item of currentLocalStorage) {
+        console.log(item);
+        //if name is found alert user yes or no to update quantity instead. if yes update extisting obj quantity with value from quantity input in form.
+        if (item.name.toLowerCase() === event.target.food.value.toLowerCase()) {
+          alert('This item already exist, Would you like to update quantity instead?');
+          updateQuantityAnswer = prompt('yes or no?');
+          if (updateQuantityAnswer.toLowerCase() === 'yes') {
+            let intNumber = (parseInt(item.quantity) + parseInt(event.target.quantity.value));
+            item.quantity = intNumber.toString();
+            console.log(`item.quantity is ${item.quantity}`);
+            break;
+          }
+
+
+
+        }
+      }
+    } else {
+      //if name isnt found push item in array.
+      currentLocalStorage.push(new PantryItem(...values));
+
+    }
+
 
-    currentLocalStorage.push(new PantryItem(...values));
     setLocalStorage("pantry", currentLocalStorage);
-  } else {
+
+  }
+
+  else {
     // no local storage yet so set it up
     this.push(new PantryItem(...values)); // `this` refers to the bound `pantryObjArray` array
 
+
     setLocalStorage("pantry", this);
   }
 
   renderFromStorage(getLocalStorage("pantry")); //currently not clearing clearing the current table
 }
 
 ///////////////// Main ///////////////////
+
 function main() {
   const form = document.getElementById("addFood");
   const storageData = getLocalStorage("pantry");
   if (storageData) {
     renderFromStorage(storageData);
   }
 
-  const pantryObjArray = []; // being put in local storage
```

Comment:
Need to comment this back in since deleting the global variables at the top of file